qcfractal/storage_sockets/mongo_socket.py

```diff
@@ -257,66 +256,58 @@ def _del_by_index(self, table, hashes, index="_id"):
 
         return (self._tables[table].delete_many({index: {"$in": hashes}})).deleted_count
 
-    def _get_generic_by_id(self, ids, table, projection=None):
+    def _get_generic(self, query, table, projection=None, allow_generic=False):
```

Comment:
Can we add a test to sift through more complex queries?

Reply:
I am not sure we should be completely happy with the current querying structure and would like to delay full tests there until we are a bit more firm that the current function is what we should be expanding on.

The names `id` and `hash` are appropriate throughout the code I believe. As `id` still refers to `table id` where `hash` refers to input hashing.

Reply:
But are we not using the `hash` as the `table id`, or is that not in yet?

Reply:
Not yet, I think we need to do more validation before that change.

Reply:
Okay, but you want to get this merged in all the same to start the process and we can add in the validation later?

Reply:
The querying structure changes and the `id->hash` conversion are two different pieces of the PR. This PR does add a `hash_index` to the results table but does not yet operationally effect anything. This is good to merge now so that we can explore hash usage IMO.

Reply:
Good enough reasoning. I'll merge it in once I clean up the pipeline.